runtime/interpreter/value.go

```diff
@@ -1136,6 +1143,14 @@ func (v *ArrayValue) handleIndexOutOfBoundsError(err error, index int, getLocati
 }
 
 func (v *ArrayValue) Get(interpreter *Interpreter, getLocationRange func() LocationRange, index int) Value {
+	if index < 0 {
+		panic(ArrayIndexOutOfBoundsError{
+			Index:         index,
+			Size:          v.Count(),
+			LocationRange: getLocationRange(),
+		})
+	}
+
 	storable, err := v.array.Get(uint64(index))
 	if err != nil {
 		v.handleIndexOutOfBoundsError(err, index, getLocationRange)
```

Comment:
isn't this `panic` redundant given the call to `handleIndexOutOfBoundsError` later?

Reply:
I thought so too, but turns out it is not.

Initially the Cadence functions here performed bounds checks. Given that atree was also performing bounds checks, we decided to remove the bounds checks from Cadence. 

However, the bounds checks in atree were not equivalent: The index type in Cadence is `int` (signed), whereas in atree, it is `uint64` (unsigned). Atree is only performing a bounds check if the index against the size of the array, but not if the index is negative. Negative indices were still handled, but indirectly: The conversion of a negative `int` to `uint64` results in a very large number, which is then out-of-bounds.

I cleaned this up by properly checking for negative integers before converting to `uint64`